src/rss_etl.py

```diff
@@ -0,0 +1,175 @@
+        self.printer_number = re_result.group(5)
+
+
+class OutputRecord:
+
+    def __init__(self, supa_con, table_name, output_dict):
+        self.supa_con = supa_con
+        self.table_name = table_name
+        self.output_dict = output_dict.copy()
+
+    def has_existing_supabase_records(self, matching_columns):
+        # TODO: This might suffice for now, but selecting every row in the table is not a good long-term solution
+        statement = self.supa_con.table(self.table_name).select("*", count="exact")
+        for column in matching_columns:
+            statement = statement.eq(column, self.output_dict[column])
+        existing_bill_records = statement.execute()
+        return existing_bill_records.count > 0
+
+
+class Extractor:
+
+    def __init__(self, supa_con, leg_body, rss_feed):
+        self.supa_con = supa_con
+        self.leg_body = leg_body
+        self.rss_feed = rss_feed
+
+    def extract_metadata_from_rss_feed(self, new_entry_count=3):
+        rss_feed = feedparser.parse(self.rss_feed)
+        for bill in rss_feed.entries:
+            if bill is None:
+                continue
+
+            bill_identifier = BillIdentifier(bill["guid"])
+            new_bill_insertion = self.create_and_attempt_to_insert_bill(bill, bill_identifier)
+            new_revision_insertion = self.create_and_attempt_to_insert_revision(bill, bill_identifier)
+            if new_bill_insertion or new_revision_insertion:
+                new_entry_count -= 1
+            if new_entry_count == 0:
+                break
+
+    def create_and_attempt_to_insert_revision(self, bill, bill_identifier: BillIdentifier) -> bool:
+        revisions_output_record = self.create_revisions_output_record(bill, bill_identifier)
+        if not revisions_output_record.has_existing_supabase_records(["revision_guid"]):
+            self.insert_new_record(revisions_output_record)
+            return True
+        return False
+
+    def create_and_attempt_to_insert_bill(self, bill, bill_identifier: BillIdentifier) -> bool:
+        bills_output_record = self.create_bill_output_record(bill, bill_identifier)
+        if not bills_output_record.has_existing_supabase_records(["legislative_id"]):
+            self.insert_new_record(bills_output_record)
+            return True
+        return False
+
+    # Method to obtain bill internal id from existing supabase row for a bill
+    def get_bill_internal_id(self, bill_identifier: BillIdentifier):
+        statement = self.supa_con.table("Bills").select("bill_internal_id").eq("legislative_id",
+                                                                               bill_identifier.bill_guid)
+        api_response = statement.execute()
+        return api_response.data[0]["bill_internal_id"]
+
+    def create_revisions_output_record(self, bill, bill_identifier: BillIdentifier):
+        revisions_output_record = OutputRecord(self.supa_con, "Revisions", REVISIONS_OUTPUT_DICT)
```

Comment:
Nit:

instead of having multiple subsequent assignments, you can use `dict.update`

E.g.
```python
        revisions_output_record.output_dict["bill_internal_id"] = self.get_bill_internal_id(bill_identifier)
        revisions_output_record.output_dict["printer_no"] = bill_identifier.printer_number
        revisions_output_record.output_dict["full_text_link"] = bill["link"]
        revisions_output_record.output_dict["publication_date"] = bill["published"]
        revisions_output_record.output_dict["revision_guid"] = bill["guid"]
        revisions_output_record.output_dict["description"] = bill["description"]
```

becomes

```python
revisions_output_record.output_dict.update({
    "bill_internal_id": self.get_bill_internal_id(bill_identifier),
    "printer_no": bill_identifier.printer_number,
    "full_text_link": bill["link"]

     #etc...

})
```


})
```

Reply:
Addressing this in my upcoming PR!